tests/dotnet/UnitTests/FileHelpers.cs

```diff
@@ -0,0 +1,27 @@
+using System.IO;
+
+#nullable enable
+
+namespace Xamarin.Tests {
+	public static class FileHelpers {
+		public static void CopyDirectory (string sourceDir, string destinationDir)
+		{
+			if (!Directory.Exists (sourceDir))
+				throw new DirectoryNotFoundException ($"Unable to copy source dir, it was not found: {sourceDir}");
+
+			if (!Directory.Exists (destinationDir))
+				Directory.CreateDirectory (destinationDir);
```

Comment:
From the documentation:

```
Any and all directories specified in path are created, unless they already exist or unless some part of path is invalid. If the directory already exists, this method does not create a new directory, but it returns a [DirectoryInfo](https://learn.microsoft.com/en-us/dotnet/api/system.io.directoryinfo?view=net-8.0) object for the existing directory.
```

No need to check its existence. The OS already does the deal.